src/components/PostPageDetail/index.js

```diff
@@ -582,6 +585,11 @@ const PostPageDetailIdComponent = (props) => {
             isBackButton={true}
             source={SOURCE_PDP}
             height={getHeightHeader()}
+            isFollow={item?.is_following_target || parentData?.is_following_target}
+            onPressFollUnFoll={() => {
+              followUnfollow(item);
+              setItem({...item, is_following_target: !item?.is_following_target});
+            }}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Here, the `isFollow` prop is being set based on whether the current item or parent data is being followed. The `onPressFollUnFoll` prop is being set to a function that calls `followUnfollow` with the current item and then updates the local state of the item to reflect the new follow status. This is a good example of optimistic UI, where the UI is updated immediately to reflect changes, before those changes are confirmed by the server.
```diff
+ isFollow={item?.is_following_target || parentData?.is_following_target}
+ onPressFollUnFoll={() => {
+   followUnfollow(item);
+   setItem({...item, is_following_target: !item?.is_following_target});
+ }}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->